solidity-v1/dashboard/src/pages/applications/ThresholdApplicationPage.jsx

```diff
@@ -0,0 +1,128 @@
+import React, { useCallback, useMemo, useState } from "react"
+import { useSelector } from "react-redux"
+import EmptyStatePage from "./EmptyStatePage"
+import { isSameEthAddress } from "../../utils/general.utils"
+import { LoadingOverlay } from "../../components/Loadable"
+import DataTableSkeleton from "../../components/skeletons/DataTableSkeleton"
+import AuthorizeThresholdContracts from "../../components/threshold/AuthorizeThresholdContracts"
+import ThresholdAuthorizationHistory from "../../components/threshold/ThresholdStakingAuthorizationHistory"
+import { MODAL_TYPES } from "../../constants/constants"
+import { useModal } from "../../hooks/useModal"
+import AuthorizeStakesBanner from "../../components/threshold/AuthorizeStakesBanner"
+
+const ThresholdApplicationPage = () => {
+  const [selectedOperator, setOperator] = useState({})
+  const { openModal } = useModal()
+  const thresholdAuthState = useSelector(
+    (state) => state.thresholdAuthorization
+  )
+
+  const authorizeContract = useCallback(
+    async (data, awaitingPromise) => {
+      const {
+        operatorAddress,
+        authorizerAddress,
+        beneficiaryAddress,
+        stakeAmount,
+      } = data
+      openModal(MODAL_TYPES.AuthorizeAndStakeOnThreshold, {
+        keepAmount: stakeAmount,
+        operator: operatorAddress,
+        beneficiary: beneficiaryAddress,
+        authorizer: authorizerAddress,
+        isAuthorized: false,
+      })
+    },
+    [openModal]
+  )
+
+  const stakeToT = useCallback(
+    async (data, awaitingPromise) => {
+      const {
+        operatorAddress,
+        authorizerAddress,
+        beneficiaryAddress,
+        stakeAmount,
+      } = data
+      openModal(MODAL_TYPES.StakeOnThresholdWithoutAuthorization, {
+        keepAmount: stakeAmount,
+        operator: operatorAddress,
+        beneficiary: beneficiaryAddress,
+        authorizer: authorizerAddress,
+        isAuthorized: true,
+      })
+    },
+    [openModal]
+  )
+
+  const thresholdAuthData = useMemo(() => {
+    const thresholdData = thresholdAuthState.authData.filter((dataObj) => {
```

Comment:
Can we rename to `unauthorizedStakes` or something like that to indicate that this array stores stakes to authorize?